website/content/en/docs/building-operators/golang/tutorial.md

```diff
@@ -92,12 +78,28 @@ controllers/memcached_controller.go
 
 This will scaffold the Memcached resource API at `api/v1alpha1/memcached_types.go` and the controller at `controllers/memcached_controller.go`.
 
-See the [API terminology doc][api_terms_doc] for details on the CRD API conventions.
+**Note:** This guide will cover the default case of a single group API. If you would like to support Multi-Group APIs see the [Single Group to Multi-Group][multigroup-kubebuilder-doc] doc.
 
-To understand the API Go types and controller scaffolding see the Kubebuilder [api doc][kb_api_doc] and [controller doc][kb_controller_doc].
+#### Understanding API’s
+
+The goal of this command is to create Custom Resource (CR) and Custom Resource Definition (CRD) resources for the Memcached Kind. This command is creating the API with the group `cache.example.com`, and version `v1alpha1`  which uniquely identifies the new CRD of the Memcached Kind. For more info see [Extend the Kubernetes API with CustomResourceDefinitions][[kubernetes-extend-api]]. 
+
+Consequently, by using the Operator SDK tool, we can create our APIs and objects that will represent our solutions on these platforms. Here will adds only a single kind of resource; however, it could have as many Kinds as needed (1…N). Basically, the CRDs are a definition of our customized Objects, and the CRs are an instance of it.
+
+#### Getting a better idea
+
+Let’s think about the classic scenario where the goal is to have an application and its database running on the platform with Kubernetes. Then, one object could represent the App, and another one could represent the DB. By having one CRD to describe the App and another one for the DB, we will not be hurting concepts such as encapsulation, the single responsibility principle, and cohesion. Damaging these concepts could cause unexpected side effects, such as difficulty in extending, reuse, or maintenance, just to mention a few.
+
+In conclusion, the App CRD will have as its controller the DB CRD. Imagine, that a Deployment and Service are required for the application run so that the App’s Controller will provide these resources in this example. Similarly, the DB’s controller will have the business logic implementation of its objects.
+
+In this way, for each CRD, one controller should be produced according to the design set by the controller-runtime.
```

Comment:
I'm not sure how this design is set by controller-runtime? you can make a controller that operators on multiple resource types

Reply:
Hi @jberkhahn,

We can create a controller that will work with multiple resource types. However, the scenario, for example, https://github.com/operator-framework/operator-sdk/issues/1383  goes against its design. Each type of resource(CR/CRD) created on the operator should have its own controller.

Reply:
Ok, that's good design practice, but this sentence still doesn't make any sense. It would be something like "In general, it's best practice to have one controller responsible for one CRD, as more complicated relationships can easily cause the reconciliation cycles to stall or hang."

Reply:
updated the text based on your suggestion 👍